File: matcomgrader/icpc_caribbean_qualifiers_2025/real_contest/e.py
"""
https://matcomgrader.com/problem/9813/exact-steps/
"""

# Accepted

def main():
    def binarySearch(low, high, steps):
        if low > high:
            return

        mid = (low + high) // 2
        if steps == 1:
            foo.add(mid)
            return
        
        binarySearch(low,mid-1, steps-1)
        binarySearch(mid+1,high, steps-1)

    test_cases = int(input())
    if test_cases > 100:
        test_cases = 100

    for _ in range(test_cases):
        foo = set()
        length, steps = [int(s) for s in input().split()]
        if length >=1 and length <= 100_000:
            binarySearch(0 , length-1, steps)
            nums = sorted(list(foo))
            if len(foo) >= 1:
                print(' '.join(([str(elem) for elem in nums])))

main()

File: matcomgrader/icpc_caribbean_qualifiers_2025/real_contest/e.cpp
#include <iostream>
#include <set>

using namespace std;

void binary_search(int low, int high, int remaining_steps, set<int>& found) {
    if (low > high) {
        return;
    }

    int mid = (low + high) / 2;

    if (remaining_steps == 1) {
        found.insert(mid);
    }
    
    binary_search(low, mid - 1, remaining_steps - 1, found);
    binary_search(mid + 1, high, remaining_steps - 1, found);
}

int main() {
    int test_cases;
    cin >> test_cases;

    for (int _t = 0; _t < test_cases; ++_t) {
        int list_size, steps;
        cin >> list_size >> steps;

        set<int> found;
        binary_search(0, list_size - 1, steps, found);

        for (auto& idx : found) {
            cout << idx << " ";
        }
        cout << endl;
    }
}
